src/Core/Primitives.idr

```diff
@@ -147,13 +155,14 @@ strSubstr [NPrimVal fc (I start), NPrimVal _ (I len), NPrimVal _ (Str str)]
     = Just (NPrimVal fc (Str (prim__strSubstr start len str)))
 strSubstr _ = Nothing
 
+
 add : Constant -> Constant -> Maybe Constant
 add (BI x) (BI y) = pure $ BI (x + y)
 add (I x) (I y) = pure $ I (x + y)
 add (B8 x) (B8 y) = pure $ B8 $ (x + y)
```

Comment:
This is wrong, it should of course do a mod there too.

Reply:
Why is this bug only showing up after the refactoring? O_O

Reply:
It didn't, the expected values in the test were wrong,  differed between the REPL tests and the :exec main test.